### PR TITLE
ChangeType.update stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json

### PR DESCRIPTION
**Pull request by fairicube-data**
{"filename": "dominant_leaf_type_10m/dominant_leaf_type_10m.json", "item_type": "stac_dist", "change_type": "Update", "user": "FAiRICUBE", "data_owner": true}

### CODE REVIEW

**Review by mari-s4e — CHANGES_REQUESTED** [review]
Some minor corrections needed. Most importantly, the extent does not look right. In EEA SDI spatial catalog is 
```
...
<gmd:westBoundLongitude>
<gco:Decimal>-56.506</gco:Decimal>
</gmd:westBoundLongitude>
<gmd:eastBoundLongitude>
<gco:Decimal>72.907</gco:Decimal>
</gmd:eastBoundLongitude>
<gmd:southBoundLatitude>
<gco:Decimal>24.284</gco:Decimal>
</gmd:southBoundLatitude>
<gmd:northBoundLatitude>
<gco:Decimal>72.665</gco:Decimal>
...
```
(https://sdi.eea.europa.eu/catalogue/copernicus/api/records/7b28d3c1-b363-4579-9141-bdd09d073fd8/formatters/xml)

File: stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        "reference_system": "3035",
+        "type": "spatial",
+        "unit": "metre",
+        "step": -10
+      },
+      "time": {
+        "extent": [],
+        "type": "temporal",
+        "values": [
+          "2018-03-01T00:00Z"
```
[the remark this answers is not in the document]
Comment (misev) [reply stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json:45]: 2012-2015 is a separate datacube as it's different 20 meter resolution (#302)

**Comment by misev** [comment]
> Some minor corrections needed. Most importantly, the extent does not look right. In EEA SDI spatial catalog is

The extent that you referenced is for a WGS84 bbox. But this data is in native CRS EPSG:3035, and has coordinates in metres and not degrees.

File: stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (misev) [reply stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json:45]: There's a summary of how these data are distributed across datacubes in [issue 68](https://github.com/FAIRiCUBE/FAIRiCUBE-Hub-issue-tracker/issues/68#issuecomment-2228324489). All years are ingested, they are just divided into different datacubes depending on matching resolution or pixel thematic categories.

**Comment by mari-s4e** [comment]
> > Some minor corrections needed. Most importantly, the extent does not look right. In EEA SDI spatial catalog is
> 
> The extent that you referenced is for a WGS84 bbox. But this data is in native CRS EPSG:3035, and has coordinates in metres and not degrees.

But these
````
  "bbox": [
    -56.5051419017044,
    24.2841770079211,
    72.90613675900907,
    58.95275109308057
  ],
```
are values in degrees, aren't they?

File: stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json [hunk 2]
```diff
@@ -0,0 +1,155 @@
+{
+  "type": "Feature",
+  "stac_version": "1.0.0",
+  "id": "dominant_leaf_type_10m",
+  "properties": {
+    "license": "other",
+    "description": "Provides at pan-European level in the spatial resolution of 10 m information on the dominant leaf type (broadleaved or coniferous)",
```
[the remark this answers is not in the document]
Comment (mari-s4e) [reply stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json]: Thanks for updating. It is basically the same, the problem is rather grammar: the previous one does not have the subject.

**Comment by misev** [comment]
> But these
> 
> ```
>   "bbox": [
>     -56.5051419017044,
>     24.2841770079211,
>     72.90613675900907,
>     58.95275109308057
>   ],
> ```
> are values in degrees, aren't they?

@mari-s4e I think you discovered a bug in the [catalog-editor](https://catalog-editor.eoxhub.fairicube.eu/) (cc @baloola). This WGS84 bbox was generated by the catalog-editor, I only entered the EPSG:3035 coordinates in it (the "x" and "y" in the json).

Our WMS service also reports a WGS84 bbox which matches the one you showed in the [previous comment](https://github.com/FAIRiCUBE/data-requests/pull/303#pullrequestreview-2199215701):

```
                <EX_GeographicBoundingBox>
                    <westBoundLongitude>-56.50514190170437</westBoundLongitude>
                    <eastBoundLongitude>72.9061049341568029565</eastBoundLongitude>
                    <southBoundLatitude>24.28417068794855834328</southBoundLatitude>
                    <northBoundLatitude>72.66326966834436</northBoundLatitude>
                </EX_GeographicBoundingBox>
```

File: stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (misev) [reply stac_dist/dominant_leaf_type_10m/dominant_leaf_type_10m.json]: Yes I agree; that one was from the [https://land.copernicus.eu/en/products/high-resolution-layer-dominant-leaf-type official data page]:

Screenshot_20240726_094005

**Comment by misev** [comment]
@baloola there is a bug because the generated northBoundLatitude is 58.95275109308057, but it should be 72.66.

The problem is likely that you transform individual coordinates which is incorrect (cc @bangph for pointing this out). This can be verified with `gdaltransform`:

```
$ gdaltransform -s_srs EPSG:3035 -t_srs EPSG:4326
Enter X Y [Z [T]] values separated by space, and press Return.
7400000 5500000
72.906136759009 58.9527510923263 0
```

It gives 58.952 like the catalog-editor and not 72.66.

The correct way is to transform the whole bbox, e.g. with `gdalwarp`:

1. Create a bbox_epsg3035.vrt file with content
```xml
<VRTDataset rasterXSize="650000" rasterYSize="460000">
  <SRS>EPSG:3035</SRS>
  <GeoTransform>900000, 10, 0, 5500000, 0, -10</GeoTransform>
  <VRTRasterBand dataType="Byte" band="1">
    <NoDataValue>0</NoDataValue>
    <SimpleSource>
      <SourceFilename relativeToVRT="1">dummy.tif</SourceFilename>
      <SourceBand>1</SourceBand>
      <SourceProperties RasterXSize="1" RasterYSize="1" DataType="Byte" BlockXSize="1" BlockYSize="1"/>
      <SrcRect xOff="0" yOff="0" xSize="1" ySize="1"/>
      <DstRect xOff="0" yOff="0" xSize="1" ySize="1"/>
    </SimpleSource>
  </VRTRasterBand>
</VRTDataset>
```
2. put whatever dummy.tif file in the same directory
3. check the bbox with gdalinfo:
```
Origin = (900000.000000000000000,5500000.000000000000000)
Pixel Size = (10.000000000000000,-10.000000000000000)
Corner Coordinates:
Upper Left  (  900000.000, 5500000.000) ( 56d30'18.51"W, 56d29' 4.75"N)
Lower Left  (  900000.000,  900000.000) ( 23d49'33.58"W, 24d17' 3.04"N)
Upper Right ( 7400000.000, 5500000.000) ( 72d54'22.09"E, 58d57' 9.90"N)
Lower Right ( 7400000.000,  900000.000) ( 40d39'45.75"E, 25d32'40.96"N)
```
4. reproject the file:
```
gdalwarp -t_srs EPSG:4326 -of VRT bbox_epsg3035.vrt bbox_epsg4326.vrt
```
5. check the result bbox with `gdalinfo bbox_epsg4326.vrt`:
```
Origin = (-56.505141901704370,72.663269668344356)
Pixel Size = (0.000128060211603,-0.000128060211603)
Corner Coordinates:
Upper Left  ( -56.5051419,  72.6632697) ( 56d30'18.51"W, 72d39'47.77"N)
Lower Left  ( -56.5051419,  24.2841707) ( 56d30'18.51"W, 24d17' 3.01"N)
Upper Right (  72.9061049,  72.6632697) ( 72d54'21.98"E, 72d39'47.77"N)
Lower Right (  72.9061049,  24.2841707) ( 72d54'21.98"E, 24d17' 3.01"N)
```

**Comment by misev** [comment]
@mari-s4e FYI I created a [separate issue](https://github.com/FAIRiCUBE/catalog/issues/31) for the bbox problem.

I don't think it's practical that I go through some 30 jsons and fix these bboxes manually - I expect this to be fixed in the catalog separately, as it probably affects already published catalog entries as well. So I suggest to ignore the bbox in this and other reviews until that issue is solved.

**Review by mari-s4e — APPROVED** [review]

**Pull request merged**